pkg/worker/client.go

```diff
@@ -120,9 +120,8 @@ func (c ScrapeAndDumpClient) scrapeAndDumpAll(ctx context.Context) (err error) {
 		if sdErr := c.scrapeAndDump(ctx, sd); sdErr != nil {
 			err = sdErr
 		}
-
 	}
-	return nil
+	return err
```

Comment:
After the go version update, the new linters caught the line above as an ineffectual assignment, so I'm _guessing_ we want to return the error if there is one, but maybe we had removed it for a reason :woman_shrugging: